src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +22,154 @@ const ActorInit Bg_Lotus_InitVars = {
     (ActorFunc)BgLotus_Init,
     (ActorFunc)BgLotus_Destroy,
     (ActorFunc)BgLotus_Update,
-    (ActorFunc)BgLotus_Draw
+    (ActorFunc)BgLotus_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Init.s")
+s32 D_80AD6D10[] = { 0x48580064 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Destroy.s")
+void BgLotus_Init(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD6830.s")
+    s32 pad;
+    s32 sp2C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD68DC.s")
+    Actor_ProcessInitChain(&this->dyna.actor, (InitChainEntry*)D_80AD6D10);
+    DynaPolyActor_Init((DynaPolyActor*)this, 1);
+    DynaPolyActor_LoadMesh(play, (DynaPolyActor*)this, &gLilyPadCol);
+    this->dyna.actor.floorHeight = BgCheck_EntityRaycastFloor5(&play->colCtx, &this->dyna.actor.floorPoly, &sp2C,
+                                                               &this->dyna.actor, &this->dyna.actor.world.pos);
+    this->unk168 = 0x60;
+    this->dyna.actor.world.rot.y = ((s32)Rand_Next() >> 0x10);
+    this->unk15C = &func_80AD68DC;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD6A88.s")
+void BgLotus_Destroy(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD6B68.s")
+void func_80AD6830(Actor* thisx) {
+    BgLotus* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Update.s")
+    f32 temp_fv1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Draw.s")
+    if (thisx->params == 0) {
+        temp_fv1 = Math_SinF(this->unk166 * (M_PI / 4)) * ((0.014f * (this->unk166 / (80.0f * 0.1f))) + 0.01f);
+        thisx->scale.x = ((1.0f + temp_fv1) * 0.1f);
+        thisx->scale.z = ((1.0f - temp_fv1) * 0.1f);
+    }
+}
+
+void func_80AD68DC(BgLotus* thisx, PlayState* play) {
```

Comment:
Please don't use thisx for the main overlay type. Use "this" for BgLotus, and use "thisx" for the base Actor type inside of the BgLotus struct only.

```suggestion
void func_80AD68DC(BgLotus* this, PlayState* play) {
```